home/tests.py

```diff
@@ -96,3 +99,100 @@ def question_tag_test_data(question_test_data):
     new_pair.tag = test_tag
     new_pair.save()
     return test_tag
+
+
+@pytest.fixture
+def question():
+    return Question(profile=Profile.objects.first(),
+                    title="Question in Math",
+                    content='How much is it 1+1?',
+                    subject=Subject.objects.first(),
+                    grade=Grade.GRADE7)
+
+
+@pytest.mark.parametrize("valid_Input", [
+    (10, 1, "Question in Math", 'How much is it 1+1?',
+     datetime(2022, 4, 7, 12, 53, 29, 4, tzinfo=pytz.UTC), 1, 4, Grade.GRADE7, 1, 10, False),
+    # User entered all the fields
+    (11, 2, "Question in History", 'How many wars Israel had?',
+     datetime(2022, 4, 7, 12, 59, 55, 4, tzinfo=pytz.UTC), 1, None, Grade.GRADE8, 2, 11, False),
+    # User entered all the fields except Sub_Subject
+    (11, 2, "Question in History", 'How many wars Israel had?',
+     datetime(2022, 4, 7, 12, 59, 55, 4, tzinfo=pytz.UTC), 1, 4, Grade.GRADE8, None, 11, False),
+    # User entered all the fields except Book
+    (10, 1, "Question in Math", 'How much is it 1+1?',
+     datetime(2022, 4, 7, 12, 53, 29, 4, tzinfo=pytz.UTC), 1, 4, Grade.GRADE7, 1, None, False),
+    # User entered all the fields except Book page
+])
+# checks if the question was successfully inserted
+@pytest.mark.django_db
+def test_add_valid_question(valid_Input):
+    question = Question(*valid_Input)
+    question.save()
+
+    assert Question.objects.filter(pk=question.id).exists()
+
+
+@pytest.mark.parametrize("invalid_Input", [
+    (10, None, "Question in Math", 'How much is it 1+1?',
+     datetime(2022, 4, 7, 12, 53, 29, 4, tzinfo=pytz.UTC), 1, 4, Grade.GRADE7, 1, 10, False),
+    # User didn't enter title
+    (10, 1, None,
+     datetime(2022, 4, 7, 12, 53, 29, 4, tzinfo=pytz.UTC), 1, 4, Grade.GRADE7, 1, 10, False),
+    # User did'nt enter content
+    (10, 1, "Question in Math", 'How much is it 1+1?', "", 1, 4, Grade.GRADE7, 1, 10, False),
+    # The question does not contain date
+    (10, 1, "Question in Math", 'How much is it 1+1?',
+     datetime(2022, 4, 7, 12, 53, 29, 4, tzinfo=pytz.UTC), "", 4, Grade.GRADE7, 1, 10, False),
+    # User didn't choose subject
+    (10, 1, "Question in Math", 'How much is it 1+1?',
+     datetime(2022, 4, 7, 12, 53, 29, 4, tzinfo=pytz.UTC), 1, 4, "", 1, 10, False),
+    # User did'nt choose grade
+    (10, 1, "Question in Math", 'How much is it 1+1?',
+     datetime(2022, 4, 7, 12, 53, 29, 4, tzinfo=pytz.UTC), 1, 4, Grade.GRADE7, 1, -10, False),
+    # User chose negative book number
+    (10, 1, "Question in Math", 'How much is it 1+1?',
+     datetime(2022, 4, 7, 12, 53, 29, 4, tzinfo=pytz.UTC), 1, 4, Grade.GRADE7, 1, 99990, False),
+    # User entered to high book number
+])
+# checks if the question was not inserted
+@pytest.mark.django_db
+def test_add_invalid_question(invalid_Input):
+    invalid = False
+
```

Comment:
To test for exceptions, please use `pytest.raises`. If different inputs are expected to raise different types of exceptions, you can pass the expected exception type is part of the "parametrize" structure.